pkg/core/resources/apis/mesh/mesh_validator.go

```diff
@@ -116,29 +133,58 @@ func validateTracingBackend(backend *mesh_proto.TracingBackend) validators.Valid
 	if backend.Name == "" {
 		verr.AddViolation("name", "cannot be empty")
 	}
+	if backend.GetType() != mesh_proto.TracingZipkinType {
+		verr.AddViolation("type", fmt.Sprintf("unknown backend type. Available backends: %q", mesh_proto.TracingZipkinType))
+	}
 	if backend.Sampling.GetValue() < 0.0 || backend.Sampling.GetValue() > 100.0 {
 		verr.AddViolation("sampling", "has to be in [0.0 - 100.0] range")
 	}
-	if zipkin, ok := backend.GetType().(*mesh_proto.TracingBackend_Zipkin_); ok {
-		verr.AddError("zipkin", validateZipkin(zipkin.Zipkin))
+	if backend.GetType() == mesh_proto.TracingZipkinType {
+		verr.AddError("config", validateZipkin(backend.Config))
 	}
 	return verr
 }
 
-func validateZipkin(zipkin *mesh_proto.TracingBackend_Zipkin) validators.ValidationError {
+func validateZipkin(cfgStr *structpb.Struct) validators.ValidationError {
```

Comment:
Here I also don't insist, that's up to you, but less nesting looks better to me:
```go
func validateZipkin(cfgStr *structpb.Struct) validators.ValidationError {
	var verr validators.ValidationError
	cfg := mesh_proto.ZipkinTracingBackendConfig{}
	if err := proto.ToTyped(cfgStr, &cfg); err != nil {
		verr.AddViolation("", fmt.Sprintf("could not parse config: %s", err.Error()))
		return verr
	}

	if cfg.ApiVersion != "" && cfg.ApiVersion != "httpJsonV1" && cfg.ApiVersion != "httpJson" && cfg.ApiVersion != "httpProto" {
		verr.AddViolation("apiVersion", fmt.Sprintf(`has invalid value. %s`, AllowedValuesHint("httpJsonV1", "httpJson", "httpProto")))
	}

	if cfg.Url == "" {
		verr.AddViolation("url", "cannot be empty")
		return verr
	} 

	uri, err := url.ParseRequestURI(cfg.Url)
	if err != nil {
		verr.AddViolation("url", "invalid URL")
	} else if uri.Port() == "" {
		verr.AddViolation("url", "port has to be explicitly specified")
	}

	return verr
}
```

Reply:
same as above